fdbserver/RestoreApplier.actor.cpp

```diff
@@ -151,85 +153,204 @@ ACTOR Future<Void> applyToDB(Reference<RestoreApplierData> self, Database cx) {
 
 	self->sanityCheckMutationOps();
 
-	state std::map<Version, Standalone<VectorRef<MutationRef>>>::iterator it = self->kvOps.begin();
-	state std::map<Version, Standalone<VectorRef<MutationRef>>>::iterator prevIt = it;
-	state int index = 0;
-	state int prevIndex = index;
-	state int count = 0;
-	state Reference<ReadYourWritesTransaction> tr(new ReadYourWritesTransaction(cx));
-	state int numVersion = 0;
+	// When the current txn fails and retries, startItInUncommittedTxn is the starting iterator in retry;
+	// startIndexInUncommittedTxn is the starting index in retry;
+	state std::map<Version, Standalone<VectorRef<MutationRef>>>::iterator curItInCurTxn = self->kvOps.begin();
+	state int curIndexInCurTxn = 0; // current index in current txn; it increases per mutation
+
+	// In case a version has 0 txns
+	while (curItInCurTxn != self->kvOps.end() && curIndexInCurTxn >= curItInCurTxn->second.size()) {
```

Comment:
Isn't it always that `curItInCurTxn->second.empty()` can tell you there is no mutations? It seems to me `curIndexInCurTxn` is always 0.

Reply:
`curIndexInCurTxn >= curItInCurTxn->second.size()` can be replaced with `curItInCurTxn ->second.empty()`